### DISCUSSION TITLE
Image size, NAXIS?

### DISCUSSION THREAD

**Discussion by Planet30** [brampf/fitscore#9] [General]
Ì noticed you are calculating the size of 2D images or hdu.dataunit.count (i.e. two axes, i.e NAXIS=2) by multiplying NAXIS * NAXIS(1) * NAXIS(2) * bits/pixel.  NAXIS is the number of axes and should not be included in this calculation... Please, let me know, if I misunderstood your code maybe.

**Comment by brampf** [comment]
Hey,

I'm actually not sure what part of the code you are referring to, so I try my best to answer this:

The `count` property of the [DataUnit](https://github.com/brampf/fitscore/blob/master/Sources/FITS/HDU/DATA/DataUnit.swift)  returns the byte count of the actual structure in memory.

If you are referring to the estimated size of the data unit based on the information provided by the header unit, this is actually computed in the property 'dataSize' of the [HeaderUnit](https://github.com/brampf/fitscore/blob/master/Sources/FITS/HDU/HEADER/HeaderUnit.swift). 

Does that answer your question?

**Reply by Planet30** [reply]
Thank you!  Figured out the size... Probably a misunderstanding from my side. I use a 16bit FITS image for testing, i.e. 2 Bytes per pixel, not 1B. In the documentation it looked like you are multiplying all values of the naxis array to calculate the datasize.
